src/clj_foundation/errors.clj

```diff
@@ -1,23 +1,37 @@
 (ns clj-foundation.errors
   (:require [clojure.tools.logging :as log]
+            [clj-foundation.patterns :refer :all]
             [schema.core :as s :refer [=> =>*]])
 
   (:import [java.util Date])
 
   (:gen-class))
 
 
-(defn replace-nil
-  [maybe-nil replacement]
-  (if (nil? maybe-nil)
-    replacement
-    maybe-nil))
+(s/defn exception-seq :- [Throwable]
+  "Returns a lazy sequence of all nested exceptions starting with the top-level exception and working to
+   the initial exception by recursively unrolling the (.getCause exception) for each nested exception.."
+  [exception :- Throwable]
+  (if exception
+    (lazy-seq (cons exception (exception-seq (.getCause exception))))))
 
 
-(defn not-nil [value name]
-  (if (nil? value)
-    (throw (java.lang.IllegalArgumentException. (str name " cannot be nil")))
-    value))
+(s/defn expect-within :- s/Any
```

Comment:
The code in there isn't really handling errors, which is what I'd expect of the module.